src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +392,402 @@ else if (!period.isEmpty()) {
         }
     }
 
+    /**
+     * Finds and return if summary tag present.
+     *
+     * @param javadoc Javadoc root node.
+     * @return true, if first sentence contains @summary tag.
+     */
+    private static boolean containsSummaryTag(DetailNode javadoc) {
+        boolean contains = false;
+        for (DetailNode node : javadoc.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
```

Comment:
body of this IF you have twice, here and at containsSummaryTagWithHtmlTag, please make method for it
and reuse.

Reply:
Done. But, I had to remove `trimWhiteSpace()` method as I am on limit of max 34 methods per class.

Reply:
this is violation from our Check, we can always suppress violations when required and reasonable.
please restore method and add this SummaryJavadocCheck.java in suppression for now.
We will think on what to do with it at the end.